src/genie/libs/parser/iosxe/show_routing.py

```diff
@@ -1916,23 +1939,51 @@ def cli(self, vrf='', output=None):
             if m:
                 vrf_dict['maximum_paths'] = int(m.groupdict()['max_path'])
                 continue
+
+            m = p3_0.match(line)
+            if m:
+                if m.groupdict()['replicates']:
+                    replicates_flag = True
+                elif m.groupdict()['no_replicates']:
+                    replicates_flag = False
+
             # application     0           0           0           0           0
             m = p3.match(line)
             if m:
-                group = m.groupdict()
-                protocol = group.pop('protocol')
-                instance = group.pop('instance')
-                if protocol == 'Total':
-                    protocol_dict = vrf_dict.setdefault('total_route_source', {})
-                else:
-                    protocol_dict = vrf_rs_dict.setdefault(protocol, {})
-                if instance is not None:
-                    inst_dict = protocol_dict.setdefault(instance, {})
-                    inst_dict.update({k:int(v) for k, v in group.items() if v is not None})
-                else:
-                    group = {k: int(v) for k, v in group.items() if v is not None}
-                    protocol_dict.update(group)
-                continue
+                if replicates_flag:
+                    group = m.groupdict()
+                    protocol = group.pop('protocol')
+                    instance = group.pop('instance')
+                    if protocol == 'Total':
+                        protocol_dict = vrf_dict.setdefault('total_route_source', {})
+                    else:
+                        protocol_dict = vrf_rs_dict.setdefault(protocol, {})
+                    if instance is not None:
```

Comment:
if instance does not do the same thing in this case?
```suggestion
                    if instance:
```

Reply:
That's correct. I'll update it to be consistent with my additions